src/wp-includes/nav-menu.php

```diff
@@ -780,6 +752,40 @@ function wp_get_nav_menu_items( $menu, $args = array() ) {
 	return apply_filters( 'wp_get_nav_menu_items', $items, $menu, $args );
 }
 
+/**
+ * Prime all linked objects to menu items.
+ *
+ * @since 6.1.0
+ *
+ * @param WP_Post[] $menu_items Array post objects of menu items.
+ */
+function update_menu_item_cache( $menu_items ) {
+	$post_ids = array();
+	$term_ids = array();
+
+	foreach ( $menu_items as $menu_item ) {
+		if ( 'nav_menu_item' !== $menu_item->post_type ) {
+			continue;
+		}
+		$object_id = get_post_meta( $menu_item->ID, '_menu_item_object_id', true );
+		$type      = get_post_meta( $menu_item->ID, '_menu_item_type', true );
+
+		if ( 'post_type' === $type ) {
+			$post_ids[] = (int) $object_id;
+		} elseif ( 'taxonomy' === $type ) {
+			$term_ids[] = (int) $object_id;
+		}
+	}
+
+	if ( ! empty( $post_ids ) ) {
+		_prime_post_caches( $post_ids, false );
+	}
```

Comment:
The case where `post_ids` is empty is already handled by `$post_ids` so I don't think is necessary to introduce a conditional.

```suggestion
	_prime_post_caches( $post_ids, false );
```

Reply:
This is just copy and pasted logic from elsewhere, I don't see any benefit in changing it. Maybe in a follow up ticket.